pybpodapi/bpod/emulator/emulator.py

```diff
@@ -0,0 +1,668 @@
+            hw_state = sma.output_matrix[i]
+            hw_state = [
+                evt
+                for evt in hw_state
+                if evt[0] < global_timer_trigger_event_pos
+            ]
+            n_differences = len(hw_state)
+            output_matrix += [n_differences]
+            for hw_conf in hw_state:
+                output_matrix += hw_conf[:2]
+        logger.debug("OUTPUT MATRIX: %s", output_matrix)
+        # Global timer start matrix (global timer-start triggered transitions)
+        global_timer_start_matrix = []
+        global_timer_start_event_pos = \
+            self.hardware.channels.events_positions.globalTimerStart
+        for i in range(sma.total_states_added):
+            state_transitions = sma.global_timers.start_matrix[i]
+            n_transitions = len(state_transitions)
+            global_timer_start_matrix += [n_transitions]
+            for transition in state_transitions:
+                dest_state = transition[1]
+                global_timer_start_matrix += [
+                    transition[0] - global_timer_start_event_pos]
+                global_timer_start_matrix += [
+                    sma.total_states_added
+                    if math.isnan(dest_state)
+                    else dest_state
+                ]
+        logger.debug("GLOBAL_TIMER_START_MATRIX: %s",
+                     global_timer_start_matrix)
+
+        # Global timer end matrix (global timer-end triggered transitions)
+        global_timer_end_matrix = []
+        global_timer_end_event_pos = \
+            self.hardware.channels.events_positions.globalTimerEnd
+        for i in range(sma.total_states_added):
+            state_transitions = sma.global_timers.end_matrix[i]
+            n_transitions = len(state_transitions)
+            global_timer_end_matrix += [n_transitions]
+            for transition in state_transitions:
+                dest_state = transition[1]
+                global_timer_end_matrix += [
+                    transition[0] - global_timer_end_event_pos
+                ]
+                global_timer_end_matrix += [
+                    sma.total_states_added if math.isnan(
+                        dest_state) else dest_state
+                ]
+        logger.debug("GLOBAL_TIMER_END_MATRIX: %s", global_timer_end_matrix)
+        # Global counter matrix (global counter triggered transitions)
+        global_counter_matrix = []
+        global_counter_event_pos = \
+            self.hardware.channels.events_positions.globalCounter
+        for i in range(sma.total_states_added):
+            state_transitions = sma.global_counters.matrix[i]
+            n_transitions = len(state_transitions)
+            global_counter_matrix += [n_transitions]
+            for transition in state_transitions:
+                dest_state = transition[1]
+                global_counter_matrix += [
+                    transition[0] - global_counter_event_pos
+                ]
```

Comment:
Bit of nit-picking, here and and other lines, since this is a single element that adds just one element above, IMO making compact looks better:
```suggestion
                    transition[0] - global_counter_event_pos]
```
However this suggestion is a matter of personal taste, so do as you desire